scripts/camera.lua

```diff
@@ -292,4 +310,59 @@ function Camera.setTransitionPeriod(camera, transitionPeriod)
     Camera.updateConfig(camera)
 end
 
+--- @param player LuaPlayer
+--- @param camera Camera.camera
+function Camera.updateChartTags(player, camera)
+    -- we can't do this without a player or force
+    if not player or not player.force then
+        return
+    end
+    
+    -- we have duplicated calculations for recovering tile coordinates from center/zoom
+    -- if we had a third box, I think it would be worth extracting this logic/data somehow
+
+    --- do the work of deleting and reinstating the tag
+    local function modifyTag(name, pos, icon)
+        local tag = camera.chart_tags[name]
+        if tag ~= nil then
+            tag.destroy()
+        end
+
+        camera.chart_tags[name] = player.force.add_chart_tag(
+            player.surface,
```

Comment:
```suggestion
            camera.surfaceName,
```

So it will show on the correct map (independent of where the player is)